projects/common/de10nano/de10nano_system_qsys.tcl

```diff
@@ -128,11 +128,11 @@ proc ad_cpu_interconnect {m_base m_port} {
   set_connection_parameter_value sys_hps.h2f_lw_axi_master/${m_port} baseAddress ${m_base}
 }
 
-proc ad_dma_interconnect {m_port} {
-
-  add_connection ${m_port} sys_hps.f2h_sdram0_data
-  set_connection_parameter_value ${m_port}/sys_hps.f2h_sdram0_data baseAddress {0x0000}
+proc ad_dma_interconnect {m_port m_id} {
 
+  # one must manually create a new(e.g. 3'rd) f2h_sdram port in order to use it
```

Comment:
We should test it the port is enabled and if not enable it. Not to mention, what happens if $m_id is not defined? We already calling this procedure without $m_id.

Reply:
This is a bot more complex, let's continue the discussion somewhere else.